### ISSUE TITLE
Crash on long-press

### ISSUE THREAD

**Issue by attah** [Karry/osmscout-sailfish#159]
Crash on long-press. Reproduces 100% of the time, tested in many locations, across a few countries. No downloaded maps.

From running OSM scout in the terminal:
```
[D] unknown:0 - long tap  QPoint(859,987)
[D] onLongTap:272 - long tap: 859x987 @ 58.12509867350767 15.296903446932012
Map DPI override: 192.00000
Map DPI override: 192.00000
[D] unknown:0 - Overlay tile cache: osmscout::TileCache(0x9ec6f4)
[D] unknown:0 - Initialize
[D] unknown:0 - ~PlaneMapRenderer
[D] unknown:0 - ~MapRenderer
harbour-osmscout: /home/mersdk/share/SailfishOS/projects/osmscout-sailfish/dependencies/libosmscout/libosmscout/src/osmscout/util/Bearing.cpp:91: std::string osmscout::Bearing::LongDisplayString() const: Assertion `false' failed.
Aborted
```
1.5 works, subsequent versions have this behaviour.
Let me know if you need more info/logs. Tried building the project, but that doesn't work for me if i do like with normal projects, so couldn't debug the value that falls outside the 0..360 range.

**Comment by Karry** [comment]
Well, I don't understand how it is possible, bearing is normalised in constructor and I am not able to reproduce.... But it should be simple to find it out... Can you please install and try to reproduce it with this version: https://osmscout.karry.cz/tests/harbour-osmscout-1.10-1.armv7hl.rpm

I just add simple debug messages around bearing computation and toString method...

**Comment by attah** [comment]
Thanks for looking in to this :) Here are the logs from that version.
```
[D] unknown:0 - long tap  QPoint(533,855)
[D] onLongTap:272 - long tap: 533x855 @ 57.486602205641915 15.0357420935211
Map DPI override: 192.00000
Map DPI override: 192.00000
[D] unknown:0 - Overlay tile cache: osmscout::TileCache(0x9c728c)
[D] unknown:0 - Initialize
[D] unknown:0 - ~PlaneMapRenderer
[D] unknown:0 - ~MapRenderer
Evaluate bearing between 0 0 -> 57.4866 15.0357: 0.163872 rad / 9.38919 deg
Bearing string for 0.163872 rad / 9.38919 deg
Evaluate bearing between -nan 0 -> 57.4866 15.0357: nan rad / nan deg
Bearing string for nan rad / nan deg
harbour-osmscout: /home/mersdk/share/SailfishOS/projects/osmscout-sailfish/dependencies/libosmscout/libosmscout/src/osmscout/util/Bearing.cpp:93: std::string osmscout::Bearing::LongDisplayString() const: Assertion `false' failed.
Aborted

```

**Comment by Karry** [comment]
Thank you! It is clear your location is NaN for some reason?! I will fix bearing for this special case... You have disabled GPS receiver?

**Comment by attah** [comment]
Yes, positioning is disabled. Enabling it does not immediately solve the issue.
The "Where am I?" menu option is greyed out, so i'll give it a while and see if it it becomes enabled and the behaviour changes.

**Comment by attah** [comment]
Weird, now the positioning appears to have started, and it still crashes.
Weirder still, the "Where am I?" menu option crashed the app too when it had become selectable.
However i did not get a log for that, and now it works, both long-press and "Where am I?".

**Comment by Karry** [comment]
It is really weird. From the log, only latitude is NaN, longitude is zero... Nevermind, application should not crash when location is not valid... It will be fixed. I will release new version in few days...

**Comment by Karry** [comment]
Should be fixed in version 1.10 that was just released. Please let me know if you will experience any troubles. Thank you again for your time!

**Comment by Karry** [comment]
Thank you for donation!
